src/generators/clientFileGenerator.ts

```diff
@@ -594,27 +594,28 @@ function getTrack2DefaultContent(
     return (
       defaultContent +
       `
+      let bearerTokenAuthenticationPolicyFound: boolean = false;
       if (options?.pipeline && options.pipeline.getOrderedPolicies().length > 0) {
         const pipelinePolicies: coreRestPipeline.PipelinePolicy[] = options.pipeline.getOrderedPolicies();
-        const bearerTokenAuthenticationPolicyFound = pipelinePolicies.some(
+        bearerTokenAuthenticationPolicyFound = pipelinePolicies.some(
           pipelinePolicy =>
             pipelinePolicy.name ===
             coreRestPipeline.bearerTokenAuthenticationPolicyName
         );
-        if (!bearerTokenAuthenticationPolicyFound) {
-          this.pipeline.removePolicy({
-            name: coreRestPipeline.bearerTokenAuthenticationPolicyName
-          });
-          this.pipeline.addPolicy(
-            coreRestPipeline.bearerTokenAuthenticationPolicy({
-              scopes: \`\${optionsWithDefaults.baseUri}/.default\`,
-              challengeCallbacks: {
-                authorizeRequestOnChallenge:
-                  coreClient.authorizeRequestOnClaimChallenge
-              }
-            })
-          );
-        }
+      }
+      if (!options || !options.pipeline || options.pipeline.getOrderedPolicies().length == 0 || !bearerTokenAuthenticationPolicyFound) {
```

Comment:
I think below change could be simpler
```suggestion
      if (!bearerTokenAuthenticationPolicyFound) {
```

Reply:
The change is not required. Refer my previous comment

Reply:
I just think this change would be simpler. Because your default value is `false` expect it would be truly if user has defined policy. So the rest of conditions would fit into add default one.

Anyway I think both are fine.